src/main/java/workers/CourseCreator.java

```diff
@@ -88,23 +93,50 @@ public static Course generateCourse(String courseId, char session) throws IOExce
      * Extracts all course names from a string containing the course exclusions and puts it into an
      * ArrayList
      *
-     * @param value a String that corresponds to all of the exclusions for a course
+     * @param value a String that corresponds to all the exclusions for a course
      * @return an ArrayList of course names
      */
     public static ArrayList<String> getCourseExclusions(String value) {
         ArrayList<String> shortenedCodes = new ArrayList<>();
         try {
-            String cleanedValue = value.replace("\"", "").replace(".", "");
-            ArrayList<String> values = new ArrayList<>(List.of(cleanedValue.split("\\s*,\\s*")));
-            for (String s: values) {
-                shortenedCodes.add(s.substring(0, 6));
-            }
-        } catch (Exception IndexOutOfBoundsException){
+            extractCodes(value, shortenedCodes);
+        } catch (Exception IndexOutOfBoundsException) {
             System.out.println("The course has no exclusions (empty string)");
         }
         return shortenedCodes;
     }
 
+    /**
+     * Extracts all course names from a string containing the course corequisites and puts it into
+     * an ArrayList
+     *
+     * @param value a String that corresponds to all the corequisites for a course
+     * @return an ArrayList of course names
+     */
+    public static List<String> getCourseCorequisites(String value) {
+        ArrayList<String> shortenedCodes = new ArrayList<>();
+        try {
+            extractCodes(value, shortenedCodes);
+        } catch (Exception IndexOutOfBoundsException) {
+            System.out.println("The course has no corequisites (empty string)");
+        }
+        return shortenedCodes;
+    }
+
+    /**
+     * Helper method for extracting the course code values from the API contents
+     *
+     * @param value a String that corresponds to all the corequisites for a course
+     * @param shortenedCodes an empty arraylist to add the course codes to
+     */
+    private static void extractCodes(String value, ArrayList<String> shortenedCodes) {
+        String cleanedValue = value.replace("\"", "").replace(".", "");
+        ArrayList<String> values = new ArrayList<>(List.of(cleanedValue.split("\\s*,\\s*")));
```

Comment:
if u make values a generic List u can just do `List<String> values = List.of(cleanedValue.split("\\s, \\s*"))`